### ISSUE TITLE
Plotting smooths (package "itsadug")

### ISSUE THREAD

**Issue by dschoenig** [QCBSRworkshops/workshop08#15]
Currently, the sections on other distribution and mixed effects rely heavily on `itsadug::plot_smooth()`, with the idea to plot smooth terms at the response scale.

I understand that this intends to make life easier for the participants of the workshop, but it actually has several problematic aspects:

- In the common case that other continuous covariates are present, their value has to be fixed, and this is not discussed in the presentation.
- Whoever works with generalized models and / or mixed models should be comfortable interpreting the effects on the *link* rather than the response scale. I think we should help participants work on that skill instead of avoiding it.
- If plots at the response scale are needed, these can be easily produced with the combination of `predict()` and `plot()`, which appears to be the standard workflow in these cases. This also works for random effects, as these can be excluded from predictions.

In short, I think the current implementation may cause more problems (for the participants) down the line than it solves.

This could be addressed when working on #14 .

**Comment by gavinsimpson** [comment]
While I agree that plotting and understanding the partial effects of GAMs is an important skill for anyone using GAMs to master, they are not easy to interpret, especially in situations where exotic link functions or families are used. As the class of models fitted by *mgcv* and related packages increases to include much more complex models (I'm thinking of location-scale models especially here), being able to evaluate what the model describes/encodes *on the response scale* is an essential skill for those using GAMs.

Why? Because, as applied scientists using statistics, we should concern ourselves with whether an estimated effect is *biologically* meaningful and the most immediate and useful way of doing that is to produce counterfactual predictions on the response scale (these are often called conditional values or predictions).

> If plots at the response scale are needed, these can be easily produced with the combination of predict() and plot(), which appears to be the standard workflow in these cases. This also works for random effects, as these can be excluded from predictions.

*easily* is hiding a lot of tedious coding there - to do this correctly, one has to

1. prepare the required data for prediction (the data used to fit the model are rarely appropriate for visualisation),
2. predict using Simon's quirky `predict()` method (what it returns depends on the arguments and can be a 1d matrix, a list of vectors or matrices, or a proper matrix),
3. form the credible intervals -- which need to be done on the link scale and then back transformed, and
4. only then can you think about plotting, which in itself also requires multiple lines of code.

This really is tedious and unnecessary. When I teach GAMs, I teach the recipe I just outlined, but then _immediately_ show how to avoid all the tedious coding using, in my case, `data_slice()` and `fitted_values()` from my *gratia* package (or now `conditional_values()`), but *itsadug* or other packages can achieve the same results (and all allow `exclude` or `terms` to be used to set one or more terms to 0 contribution (e.g. for ranefs).

**Comment by dschoenig** [comment]
I agree, those are all very good points @gavinsimpson 

Biologically, showing effects at the response scale is immensely important. When teaching, I also follow the same workflow you outlined, and then explain how to use `gratia` to produce plots of conditional effects, and then move to `marginaleffects` & `ggplot2` for marginal effects.

I haven't contributed to the material of this workshop in a while, but I think `gratia` could be used more, since it also includes a user-friendly way to simulate from the approximate posterior predictive distribution (which again works with vanilla `mgcv` but is another multi-step procedure).

What I generally see is that people learning about these models need a workable understanding of how conditional and marginal effects are calculated in order to interpret the corresponding plots at the response scale well (particularly in the case of interactions, random effects, and non-collapsible link functions). In ecology, at least, I feel that there is a shift away from interpreting regression parameters towards interpreting effects at the response scale (which I find positive), and I wonder whether it would make sense for QCBS to set up a new workshop only based on generating and interpreting conditional and marginal effects plots from GLMs and GAMs. The `marginaleffects` documentation is quite good but still daunting for someone how has little prior knowledge.

Possible actions:
- The interactive workshop / slides maybe leaning more heavily on a package with specialized plotting capabilities would be a good choice (personally I would  prefer this package to `itsadug` because of the interface other included functionality, but opinions might differ), while in the "book" version we could show how to achieve the same with `mgcv` only.
- I think learning t distinguish between link and response scale is a worthwhile learning object, but it is one that applies to generalized models more broadly, and not only to GAMs, so maybe it makes sense to give it a lower priority for the material of this particular workshop, and think about adding one specifically for conditional and marginal effect *interpretation*.
